src/app/Instance/CreateInstance/CreateInstance.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FormEvent, useCallback, useEffect, useState } from "react";
+import {
+  Alert,
+  AlertGroup,
+  Button,
+  Form,
+  FormGroup,
+  Modal,
+  TextInput,
+} from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+
+export interface CreateInstanceProps {
+  /**
+   * Flag to indicate the creation request is in progress
+   */
+  isLoading: boolean;
+  /**
+   * Flag to show/hide the modal
+   */
+  isModalOpen: boolean;
+  /**
+   * Callback to close the modal
+   */
+  onClose: () => void;
+  /**
+   * Callback to create the instance
+   */
+  onCreate: (name: string) => void;
+}
+
+const CreateInstance = (props: CreateInstanceProps) => {
+  const { isLoading, isModalOpen, onClose, onCreate } = props;
+  const [name, setName] = useState("");
+  // @TODO Basic temporary error management. To be integrated with APIs errors.
+  const [error, setError] = useState<string | null>(null);
+  const { t } = useTranslation("openbridgeTempDictionary");
+
+  const FORM_ID = "create-instance-form";
+
+  const onSubmit = useCallback(
+    (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (name === "") {
+        setError("name-required");
+      } else {
+        setError(null);
+        onCreate(name);
+      }
+    },
+    [name, onCreate]
+  );
+
+  const handleNameChange = useCallback((value: string) => {
+    setName(value);
+    setError(value === "" ? "name-required" : null);
```

Comment:
Is there some App Services related explanation for this magic string and subsequent use on L#59?

Reply:
No. this is only a magic string.  I added a comment on L#35 about it. It's only a temporary basic validation. I think the real one will be plugged to the APIs and will refer to error codes coming from there. So we'll reuse the "name missing" error code when preventing the form submission when the name input is empty.
Also maybe the validation will be handled outside the `CreateInstance` component and passed down as a prop? We have to discuss it properly.